shared/lib/routes/example/example.js

```diff
@@ -1,27 +1,30 @@
-/*global JS_ENV Promise require*/
+/*global JS_ENV Map Promise require*/
 
 import RouteBase from './../route.base';
 
 class ExampleRoute extends RouteBase {
 
-  get route_name(){
+  get route_name() {
     return 'ExampleRoute';
   }
 
-  assureData(state){
+  assureData(state) {
     let route = this;
-
-    if (!route.stateExampleSet(state)){
+    if (!this.repo) {
       let ExampleRepo = require(`./../../../../${JS_ENV}/models/example/example.repository.js`);
-      return ExampleRepo.findById(route.params.example_id)
-        .then((example)=>{
+      this.repo = new ExampleRepo(new Map());
+    }
+
+    if (!route.stateExampleSet(state)) {
+      return this.repo.findById(route.params.example_id)
+        .then((example) => {
           state.example = example;
         });
     }
     return Promise.resolve();
   }
 
-  stateExampleSet(state){
+  stateExampleSet(state) {
     return state.example && parseInt(this.params.example_id) === parseInt(state.example.data.id)
   }
 
```

Comment:
The caching you propose is misplaced and this is a poor man's cache.  It does not provide expiry mechanism, and there is no way to actually refresh data if I need them (eg. if I hypothetically add another Example on the server side - my whole app during the session will not be able to get the new example since it will have cached results in the Map). 

So this way it is more flexible. If we need the functionality you mentioned - we can instantiate repository with already populated map that was stored elsewhere (maybe create a getter for it) and get it after first repository populated it. 

But there are other places more suitable for caching.
1. place is the browser cache - just by adding ETAG header on server responses we can use information from browser's cache if it is not changed on the server, or get and cache information if the new information has arrived on the server without changing a line of code in the current setup. This is most correct form since it will guarantee that we don't have stale data. Of course it puts more strain on the server - but we don't have the measurement to conclude if the load will be too much. 
If measurements say that it is too much for the server then: 

2. place is the API classes - we can cache responses in lokijs or localstorage directly and then return it from there if they did not expire - or depending on library we use for ajax - maybe it has caching functionality already - but of course this way client does not know if data has changed (except maybe time based expiration) ... 

We can also combine these two approaches - optimistically returned cached data from 2. and query the server like 1. after it - and if it returns new data, update the in cache.

Botom line @Arbolista  - I think that caching should be done somewhere bellow the repositories and not inside of them.